File: src/main/java/dev/slimevr/vr/trackers/TrackerStatus.java
package dev.slimevr.vr.trackers;

public enum TrackerStatus {

	DISCONNECTED(0, false),
	OK(1, true),
	BUSY(2, true),
	ERROR(3, false),
	OCCLUDED(4, false);

	private static final TrackerStatus[] byId = new TrackerStatus[5];

	static {
		for (TrackerStatus st : values())
			byId[st.id] = st;
	}

	public final int id;
	public final boolean sendData;

	TrackerStatus(int id, boolean sendData) {
		this.sendData = sendData;
		this.id = id;
	}

	public static TrackerStatus getById(int id) {
		if (id < 0 || id >= byId.length)
			return null;
		return byId[id];
	}
}

File: server/src/main/java/dev/slimevr/vr/trackers/TrackerStatus.kt
package dev.slimevr.vr.trackers

enum class TrackerStatus(
	val id: UByte,
	@JvmField val sendData: Boolean,
) {
	DISCONNECTED(0u, false),
	OK(1u, true),
	BUSY(2u, true),
	ERROR(3u, false),
	OCCLUDED(4u, false),
	;

	companion object {
		@JvmStatic
		fun getById(id: UByte): TrackerStatus? = byId[id]
	}
}

private val byId = TrackerStatus.values().associateBy { it.id }
